qt/widgets/common/src/MWRunFiles.cpp

```diff
@@ -510,65 +511,101 @@ void MWRunFiles::clear() {
   m_uiForm.fileEditor->setText("");
 }
 
+/**
+* Finds the files if the user has changed the parameter text.
+*/
+void MWRunFiles::findFiles() { findFiles(m_uiForm.fileEditor->isModified()); }
+
 /**
 * Finds the files specified by the user in a background thread.
 */
-void MWRunFiles::findFiles() {
+void MWRunFiles::findFiles(bool isModified) {
+  auto searchText = m_uiForm.fileEditor->text();
+  const auto directory = findFilesDirectory(searchText);
```

Comment:
I think there is a subtle difference in the behaviour of the old and new code here. In the old code there was an early return inside the [`if(m_isDirectory)`](https://github.com/mantidproject/mantid/blob/647f2e8bb03e4000313121d9ec580fbb379707c2/qt/widgets/common/src/MWRunFiles.cpp#L528) so regardless of the `searchText` result it would return at this point. As far as I can tell in this code  `findFilesDirectory` returns the `searchText` only if it is not empty. So the `if(directory)` check in the case where `searchText` is empty will not cause an early return anymore.